metadata-io/src/main/java/com/linkedin/metadata/timeline/eventgenerator/EntityChangeEventGenerator.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedin.metadata.timeline.eventgenerator;
+
+import com.datahub.authentication.Authentication;
+import com.github.fge.jsonpatch.JsonPatch;
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.entity.EntityAspect;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.timeline.data.ChangeCategory;
+import com.linkedin.metadata.timeline.data.ChangeEvent;
+import com.linkedin.metadata.timeline.data.ChangeTransaction;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+
+public abstract class EntityChangeEventGenerator<T extends RecordTemplate> {
```

Comment:
Nice! Minor -> I'd prefer to see a java doc on this class explaining what it's for -- to make the next person's life just a bit easier!

Reply:
Added :)

Reply:
tyvm!